### PR TITLE
Update dpnp.linalg.inv() function

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR update `dpnp.linalg.inv()` function to calculate the (multiplicative) inverse matrix using `oneapi::mkl::lapack::gesv` for 2d input matrix and `oneapi::mkl::lapack::getrf_batch` and `oneapi::mkl::lapack::getri_batch`  for input array dimensions > 2.
The implementation is written as a pybind11 extension above required LAPACK functions.
The changes are related to support of all types and support of function arguments as full_matrices and compute_uv

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/linalg/dpnp_utils_linalg.py [hunk 1]
```diff
@@ -740,6 +770,147 @@ def dpnp_eigh(a, UPLO):
...
+
+    a_usm_arr = dpnp.get_usm_ndarray(a)
+    a_sycl_queue = a.sycl_queue
+    a_usm_type = a.usm_type
+
+    a_order = "C" if a.flags.c_contiguous else "F"
+    a_shape = a.shape
+
+    # oneMKL LAPACK gesv overwrites `a` and assumes fortran-like array as input.
+    # Allocate 'F' order memory for dpnp arrays to comply with these requirements.
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply dpnp/linalg/dpnp_utils_linalg.py:882]: I always allocate 'F' order memory for `gesv`

File: dpnp/linalg/dpnp_utils_linalg.py [hunk 2]
```diff
@@ -441,6 +442,35 @@ def _lu_factor(a, res_type):
...
+        Default: `Singular matrix`
+
+    Raises
+    ------
+    dpnp.linalg.LinAlgError
+        On non-zero elements in dev_info, indicating LAPACK errors.
+
+    """
+
+    dev_info_array = dpnp.array(dev_info)
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply dpnp/linalg/dpnp_utils_linalg.py:466]: You probably meant to use `any`

File: dpnp/linalg/dpnp_utils_linalg.py [hunk 3]
```diff
@@ -815,39 +809,40 @@ def dpnp_inv_batched(a, res_type):
...
 
-    ht_lapack_ev_1.wait()
-    ht_lapack_ev.wait()
-    a_ht_copy_ev.wait()
+    _check_lapack_dev_info(dev_info)
 
-    check_lapack_dev_info(dev_info_getrf_h)
-    check_lapack_dev_info(dev_info_getri_h)
+    ht_getrf_ev.wait()
+    ht_getri_ev.wait()
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_utils_linalg.py:844]:
```suggestion
    ht_getri_ev.wait()
    ht_getrf_ev.wait()
```

File: dpnp/linalg/dpnp_utils_linalg.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (antonwolfy) [reply dpnp/linalg/dpnp_utils_linalg.py:877]: no, you have `a_order` as "C" also, but in that case you pass transposed array to `gesv`. I'd suggest to rephrase the comment to be more clear around here.
```python
a_order = "C" if a.flags.c_contiguous else "F"
```

[the remark this answers is not in the document]
Comment (antonwolfy) [reply dpnp/linalg/dpnp_utils_linalg.py:877]: no, you have `a_order` as "C" also, but in that case you pass transposed array to `gesv`. I'd suggest to rephrase the comment to be more clear around here.

[the remark this answers is not in the document]
Comment (antonwolfy) [reply dpnp/linalg/dpnp_utils_linalg.py:877]: The comment is still not addressed

**Review by antonwolfy — APPROVED** [review]
LGTM! Thank you @vlad-perevezentsev

**Pull request merged**